src/Network/Socket.php

```diff
@@ -326,11 +326,11 @@ public function write($data)
         for ($written = 0; $written < $totalBytes; $written += $rv) {
             $rv = fwrite($this->connection, substr($data, $written));
             if ($rv === false || $rv === 0) {
-                return $written;
+                return (bool)$written;
```

Comment:
It would be safer to do the casting in the calling method instead of changing return type of this method.